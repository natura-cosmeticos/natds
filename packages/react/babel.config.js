const path = require('path')

const defaultAlias = {
  '@naturacosmeticos/natds-react': path.resolve(__dirname, './src/components')
}

module.exports = {
  plugins: [
    ['babel-plugin-module-resolver', { alias: defaultAlias }]
  ],
  presets: [
    ['@babel/preset-env', { targets: { node: 'current' } }],
    '@babel/preset-typescript',
    '@babel/preset-react'
  ]
}
