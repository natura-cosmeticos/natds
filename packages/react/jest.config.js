const base = require('../../jest.config.base.js')
const { displayName, name } = require('./package.json')

module.exports = {
  ...base,
  collectCoverageFrom: [
    'src/**/*.{js,jsx,ts,tsx}',
    '!src/**/*.stories.tsx',
    '!src/helpers/**'
  ],
  errorOnDeprecated: true,
  displayName,
  name,
  roots: ['<rootDir>/src'],
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
    '^.+\\.mdx?$': '@storybook/addon-docs/jest-transform-mdx'
  }
}
