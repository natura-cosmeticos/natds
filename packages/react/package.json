{
  "name": "@naturacosmeticos/natds-react",
  "displayName": "NatDS-React",
  "description": "A collection of components from Natura Design System for React",
  "version": "1.8.0-alpha.DSY-1959.4.0",
  "private": false,
  "keywords": [
    "design-system",
    "natura",
    "react",
    "web"
  ],
  "homepage": "https://github.com/natura-cosmeticos/natds-js/tree/master/packages/react/README.md",
  "bugs": {
    "url": "https://github.com/natura-cosmeticos/natds-js/issues"
  },
  "license": "ISC",
  "author": {
    "email": "designsystem@natura.net",
    "name": "Natura&Co.",
    "url": "https://github.com/natura-cosmeticos"
  },
  "files": [
    "README.md",
    "dist/"
  ],
  "jsdelivr": "dist/umd/index.js",
  "source": "src/index.ts",
  "types": "dist/index.d.ts",
  "typings": "dist/index.d.ts",
  "umd:main": "dist/umd/index.js",
  "unpkg": "dist/umd/index.js",
  "main": "dist/index.js",
  "directories": {
    "lib": "dist"
  },
  "repository": {
    "directory": "packages/react",
    "type": "git",
    "url": "git+https://github.com/natura-cosmeticos/natds-js.git"
  },
  "scripts": {
    "build": "yarn clean && tsc -p ./tsconfig.json && yarn rollup:build",
    "build:storybook": "yarn build-storybook -o docs/dist",
    "build:storybook:ci": "yarn build:storybook --quiet",
    "clean": "rm -rf ./dist",
    "copy:assets": "mkdir -p ./dist/assets && cp -r ./src/assets/* ./dist/assets",
    "lint": "eslint --ext .js,.jsx,.ts,.tsx ./src --color",
    "lint:fix": "yarn lint --fix",
    "rollup:build": "rollup -c",
    "storybook": "start-storybook -p 6006",
    "test": "jest --verbose",
    "test:ci": "jest --coverage && codecov -t",
    "test:watch": "jest --watch"
  },
  "dependencies": {
    "@naturacosmeticos/natds-icons": "1.3.0",
    "@naturacosmeticos/natds-themes": "^0.4.12",
    "react-jss": "10.5.1"
  },
  "devDependencies": {
    "@types/react": "17.0.3",
    "@types/react-dom": "17.0.3",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-test-renderer": "17.0.2"
  },
  "peerDependencies": {
    "@types/react": ">= 16.8.6",
    "@types/react-dom": "17.0.3",
    "react": ">= 16.8.4",
    "react-dom": ">= 16.8.4"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    },
    "@types/react-dom": {
      "optional": true
    }
  }
}
