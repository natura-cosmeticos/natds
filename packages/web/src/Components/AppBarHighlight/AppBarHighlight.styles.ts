import createStyles from '@material-ui/core/styles/createStyles'
import makeStyles from '@material-ui/core/styles/makeStyles'

const styles = createStyles({
  root: {
    flexGrow: 1
  }
})

export const useStyles = makeStyles(styles, { name: 'NatDSAppBarHighlight' })

export default useStyles
