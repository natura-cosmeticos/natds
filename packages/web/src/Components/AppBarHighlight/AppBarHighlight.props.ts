import * as React from 'react'

export interface AppBarHighlightProps {

  /**
   * Used to render elements inside the `AppBarHighlight`.
   * This can be an `element` like `<Typography />`.
   *
   * @required
   * @type React.ReactNode
   */
  children: React.ReactNode,

  /**
   * Specifies one or more class names for the element.
   *
   * Will be forwarded to the HTML element as the `class` attribute.
   *
   * @optional
   */
  className?: string,

  /**
   * The component used for the root node.
   * Either a string to use a HTML element of a component.
   *
   * Defaults to `div`.
   *
   * @optional
   * @type React.ElementType
   */
  component?: React.ElementType,

}
