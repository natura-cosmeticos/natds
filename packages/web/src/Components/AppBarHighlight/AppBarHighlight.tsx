import * as React from 'react'
import clsx from 'clsx'
import { AppBarHighlightProps } from './AppBarHighlight.props'
import useStyles from './AppBarHighlight.styles'

export { AppBarHighlightProps } from './AppBarHighlight.props'

export const AppBarHighlight : React.FunctionComponent<AppBarHighlightProps> = (props: AppBarHighlightProps) => {
  const {
    className, component: Component = 'div', children, ...otherProps
  } = props

  const { root } = useStyles()

  const classNames: string = clsx(
    'NatDSAppBarHighlight',
    root,
    className
  )

  return (
    <Component className={classNames} {...otherProps}>
      {children}
    </Component>
  )
}

export default AppBarHighlight
