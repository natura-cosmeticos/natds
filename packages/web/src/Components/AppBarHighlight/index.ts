import AppBarHighlight from './AppBarHighlight'

export { AppBarHighlightProps } from './AppBarHighlight.props'
export { AppBarHighlight }
export default AppBarHighlight
