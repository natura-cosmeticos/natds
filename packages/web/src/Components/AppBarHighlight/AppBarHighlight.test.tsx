import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'
import { AppBarHighlight } from './AppBarHighlight'

describe('AppBarHighlight', () => {
  test('should match snapshot - default props', () => {
    const testRenderer = TestRenderer.create(<AppBarHighlight>any children</AppBarHighlight>)

    expect(testRenderer).toMatchSnapshot()
  })
})
