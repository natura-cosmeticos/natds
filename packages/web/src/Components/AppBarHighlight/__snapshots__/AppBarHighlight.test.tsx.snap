// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBarHighlight should match snapshot - default props 1`] = `
<div
  className="NatDSAppBarHighlight NatDSAppBarHighlight-root-1"
>
  "any children"
</div>
`;
